packages/main/src/AINoticeIndicator.ts

```diff
@@ -0,0 +1,201 @@
+ *
+ * `import "@ui5/webcomponents/dist/AINoticeIndicator.js";`
+ * @constructor
+ * @extends UI5Element
+ * @since 2.12.0
+ * @public
+ */
+@customElement({
+	tag: "ui5-ai-notice-indicator",
+	renderer: jsxRenderer,
+	template: AINoticeIndicatorTemplate,
+	styles: AINoticeIndicatorCss,
+	dependencies: [
+		Label,
+		Popover,
+		Icon,
+		Button,
+		Link,
+	],
+})
+/**
+ * Fired on mouseup, space and enter if ai-notice is interactive
+ *
+ * **Note:** The event will not be fired if the `disabled`
+ * property is set to `true`.
+ * @public
+ */
+@event("click")
+class AINoticeIndicator extends UI5Element {
+	eventDetails!: {
+		click: { targetRef: HTMLElement },
+	}
+	/**
+	 * Link text for AI Notice Control
+	 *
+	 * @public
+	 * @default ""
+	*/
+	@property()
+	attributionText: string = "";
+
+	/**
+	 * Description text for AI Notice Control
+	 *
+	 * @public
+	 * @default ""
+	*/
+	@property()
+	verificationText: string = "";
+
+	/**
+	 * Show icon for AI Notice Control link text
+	 *
+	 * @public
+	 * @default false
+	*/
+	@property({ type: Boolean })
+	showIcon: boolean = false;
+
+	/**
+	 * Enable/Disable action on click - showing popup
+	 *
+	 * @public
+	 * @default false
+	*/
+	@property({ type: Boolean })
+	disabled = false;
+
+	/**
+	 * Determines how <code>AINoticeIndicator</code> will be displayed.
+	 *
+	 * @public
+	 * @default "Default"
+	*/
+	@property()
+	displayMode: `${AINoticeIndicatorDisplayMode}` = AINoticeIndicatorDisplayMode.Default;
+
+	/**
+	 * Defines the popover of the component.
+	 * @public
+	 */
+	@slot({ type: HTMLElement, "default": true })
+	_popover!: Array<HTMLElement>;
+
+	_controlNeededWidth: number = 0;
+	_onResize!: ResizeObserverCallback;
+
+	constructor() {
+		super();
+
+		this._onResize = this._handleResize.bind(this);
+	}
+
+	_handleResize() {
+		if (!this.shadowRoot) {
+			return;
+		}
+
+		const label = this.shadowRoot.querySelector<HTMLElement>("ui5-label");
+
+		if (!label) {
+			return;
+		}
+
+		const containerWidth = this.offsetWidth;
+		const margin = 2;
+
+		// Compare total width to container width
+		if (this._controlNeededWidth + margin > containerWidth) {
+			label.style.display = "none";
+		} else {
+			label.style.display = "";
+		}
+	}
+
+	_getNeededWidthToDisplayControlFully = () => {
+		if (!this.shadowRoot) {
+			return;
+		}
+
+		const button = this.shadowRoot.querySelector<HTMLElement>("ui5-button");
+		const link = this.shadowRoot.querySelector<HTMLElement>("ui5-link");
+		const label = this.shadowRoot.querySelector<HTMLElement>("ui5-label");
+
+		if (!label) {
+			return;
+		}
+
+		// Get widths
+		const buttonWidth = button?.offsetWidth || 0;
+		const linkWidth = link?.offsetWidth || 0;
+		const labelWidth = label?.offsetWidth || 0;
+
+		this._controlNeededWidth = buttonWidth + linkWidth + labelWidth;
+	}
+
+	_onLinkPress(e: UI5CustomEvent<Button | Link, "click">) {
+		e.stopPropagation();
+		this.fireDecoratorEvent("click", { targetRef: e.target as HTMLElement });
+	}
```

Comment:
Do we need to expose targetRef, can't developer just do:
`	popup.opener = e.target as HTMLElement;
	popup.open = true;
	`
	in the handler ?